rca/wagtailapi/filters.py

```diff
@@ -13,3 +17,50 @@ def filter_queryset(self, request, queryset, view):
             queryset = queryset.filter(degree_level__in=pks)
 
         return queryset
+
+
+class SearchFilter(filters.SearchFilter):
+    def filter_queryset(self, request, queryset, view):
+        """
+        This overrides the wagtail core api filters.SearchFilter
+        so we can provide autocomplet/fuzzy text matching with
+        sb.autocomplete
+        """
+        search_enabled = getattr(settings, "WAGTAILAPI_SEARCH_ENABLED", True)
+
+        if "search" in request.GET:
+            if not search_enabled:
+                raise BadRequestError("search is disabled")
+
+            # Searching and filtering by tag at the same time is not supported
+            if getattr(queryset, "_filtered_by_tag", False):
+                raise BadRequestError(
+                    "filtering by tag with a search query is not supported"
+                )
+
+            search_query = request.GET["search"]
+            search_operator = request.GET.get("search_operator", None)
+            order_by_relevance = "order" not in request.GET
+
+            sb = get_search_backend()
+            try:
+                queryset = sb.autocomplete(
```

Comment:
This is not a concern, but I wonder if this will only search field indexed with `index.AutocompleteField` and `index.SearchField`.